### PR TITLE
Centralise `echoerr` function [DI-343]

### PR DESCRIPTION
**Pull request by JackPGreen**
Currently is duplicated and should instead be centralised.

[Raised from PR feedback](https://github.com/hazelcast/hazelcast-docker/pull/823#discussion_r1865712579).

Relates to https://github.com/hazelcast/hazelcast-docker/pull/842.

### CODE REVIEW

File: check-hazelcast-health.sh [hunk 1]
```diff
@@ -1,17 +1,9 @@
 #!/usr/bin/env bash
 
-set -o errexit
-# TODO REMOVE
-set -x
+set -o errexit ${RUNNER_DEBUG:+-x}
```
[the remark this answers is not in the document]
Comment (JackPGreen) [reply check-hazelcast-health.sh:3]: I don’t think it’s consistent in Docker, either - ie between long or short form arguments.

File: .github/workflows/publish-brew-package.yml [hunk 2]
```diff
@@ -38,6 +38,12 @@ jobs:
       - name: Checkout hazelcast-packaging repo
         uses: actions/checkout@v4
 
+      - name: Checkout hazelcast-docker repo
+        uses: actions/checkout@v4
+        with:
+          repository: hazelcast/hazelcast-docker
+          path: 'hazelcast-docker'
```
Comment (ldziedziul) [inline .github/workflows/publish-brew-package.yml:45]: Have mixed feeling cloning this repo. Maybe at least checkout only a single file:
```yaml
sparse-checkout: .github/scripts/abstract-simple-smoke-test.sh
sparse-checkout-cone-mode: false
```
And then maybe extract it to composite action and replace duplicates?

Comment (JackPGreen) [reply .github/workflows/publish-brew-package.yml:45]: > Have mixed feeling cloning this repo. Maybe at least checkout only a single file:

But `abstract-simple-smoke-test.sh` `source`s `logging.functions.sh`, so it needs that as well.
And then `hazelcast-packaging` has even tighter coupling on the implementation.

Comment (ldziedziul) [reply .github/workflows/publish-brew-package.yml:45]: I see, right

File: check-hazelcast-health.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (nishaatr) [reply check-hazelcast-health.sh:3]: ok. may be something for later

**Review by nishaatr — APPROVED** [review]

**Review by ldziedziul — APPROVED** [review]

**Pull request merged**